src/core/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,934 @@
+  thumb.focus();
+  expect(thumb.getAttribute('aria-valuenow')).toEqual('50');
+
+  act(() => {
+    fireEvent.keyDown(thumb, { key: 'ArrowLeft' });
+  });
+  thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb.getAttribute('aria-valuenow')).toEqual('50');
+});
+
+it('thumb should show tooltip on hover', () => {
+  let element: HTMLDivElement | null = null;
+  const onRef = (ref: HTMLDivElement) => {
+    element = ref;
+  };
+  const wrapper = render(<Slider ref={onRef} values={defaultSingleValue} />);
+  const { container } = wrapper;
+  assertBaseElement(container);
+  expect(element).toBeTruthy();
+  let thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb).toBeTruthy();
+  expect(thumb.classList.contains('iui-active')).toBeFalsy();
+  expect(container.querySelector('.iui-tooltip')).toBeFalsy();
+  act(() => {
+    fireEvent.mouseEnter(thumb);
+  });
+  expect(
+    (container.querySelector('.iui-tooltip') as HTMLDivElement).textContent,
+  ).toBe('50');
+  act(() => {
+    thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+    fireEvent.mouseLeave(thumb);
+  });
+});
+
+it('thumb should show tooltip on focus', () => {
+  let element: HTMLDivElement | null = null;
+  const onRef = (ref: HTMLDivElement) => {
+    element = ref;
+  };
+  const wrapper = render(<Slider ref={onRef} values={defaultSingleValue} />);
+  const { container } = wrapper;
+  assertBaseElement(container);
+  expect(element).toBeTruthy();
+  let thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb).toBeTruthy();
+  expect(thumb.classList.contains('iui-active')).toBeFalsy();
+  expect(container.querySelector('.iui-tooltip')).toBeFalsy();
+  act(() => {
+    thumb.focus();
+  });
+  expect(
+    (container.querySelector('.iui-tooltip') as HTMLDivElement).textContent,
+  ).toBe('50');
+  act(() => {
+    thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+    thumb.blur();
+  });
+});
+
+it('thumb props should be applied', () => {
+  const thumbProps = () => {
+    return {
+      className: 'thumb-test-class',
+      style: { backgroundColor: 'red' },
+    };
+  };
+  const wrapper = render(
+    <Slider values={defaultSingleValue} thumbProps={thumbProps} />,
+  );
+  const { container } = wrapper;
+  assertBaseElement(container);
+  const thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb).toBeTruthy();
+  expect(thumb.classList.contains('thumb-test-class'));
+});
+
+it('focus on specific thumb', () => {
+  const wrapper = render(<Slider values={[50, 80]} step={5} />);
+  const { container } = wrapper;
+  assertBaseElement(container);
+  const slideContainer = container.querySelector(
+    '.iui-slider-container',
+  ) as HTMLDivElement;
+  let thumbs = container.querySelectorAll('.iui-slider-thumb');
+
+  focusThumb(slideContainer, 1);
+  focusThumb(slideContainer, 1);
```

Comment:
why twice?

Reply:
For coverage...to cover the line that skips processing of element is already focused.

Reply:
amm... Would make sense if you could check something (that processing was skipped). Without it, this is really hard to understand. Write tests to test scenarios, not for the coverage numbers would be a good approach.

Reply:
I am glad to remove it, I can also add a comment of why I called it twice. I do not know a good way to test it and have better things to do with my time.